lib/hexpm/web/controllers/password_controller.ex

```diff
@@ -28,6 +28,8 @@ defmodule Hexpm.Web.PasswordController do
     case Users.password_reset_finish(username, key, params, revoke_all_keys?, audit: audit_data(conn)) do
       :ok ->
         conn
+        |> clear_session()
+        |> configure_session(renew: true)
```

Comment:
Make sure you are also doing this on sign in. You need to renew every time you sign in.